src/migration/1647526647810-init.ts

```diff
@@ -99,11 +99,22 @@ export class init1647526647810 implements MigrationInterface {
             $$;
         `);
         await queryRunner.query(`
-            create or replace function iasql_install(_mods text[]) returns void
+            create or replace function iasql_install(_mods text[]) returns table (
+              module_name character varying,
+              table_name character varying,
+              record_count int
+            )
             language plpgsql
             as $$
             begin
                 perform until_iasql_operation('INSTALL', _mods);
+                return query select
+                  m.name as module_name,
+                  t.table as table_name,
+                  (xpath('/row/c/text()', query_to_xml(format('select count(*) as c from public.%I', t.table), FALSE, TRUE, '')))[1]::text::int AS record_count
```

Comment:
what is this line doing? looking at the logs in a file?

Reply:
I found the pattern here: https://stackoverflow.com/a/2611745

It's a trick to be able to do a query where the table name itself is dynamic by abusing Postgres' XML functions, similar in spirit to your "DB connects to itself as a foreign database" trick.